.github/workflows/native-image.yaml

```diff
@@ -1,42 +1,65 @@
-name: native-image
-on: [push, pull_request]
-jobs:
-  build-binary-unix:
-    runs-on: ${{ matrix.os }}
-    strategy:
-      fail-fast: true
-      matrix:
-        os: [macos-latest, ubuntu-latest]
-        include:
-          - os: macos-latest
-            label: 'darwin'
-          - os: ubuntu-latest
-            label: 'linux'
-    steps:
-    - uses: actions/checkout@v2
-    - uses: DeLaGuardo/setup-graalvm@8bbfe44ef9c6f5c07e5af036a1bffd561c037d18
-      with:
-        graalvm-version: '20.3.0.java8'
-    - run: ./mvnw package -Dnative -DskipTests $([ $(uname -s) = Linux ] && echo "-Dgraalvm.static=--static") -Dcbi.jarsigner.skip=true
-    - run: mv org.eclipse.lemminx/target/lemminx-* lemminx-$(git rev-parse --short "$GITHUB_SHA")-${{ matrix.label }}
-    - uses: actions/upload-artifact@v2
-      with:
-        name: lemminx-${{ matrix.label }}
-        path: lemminx-*-${{ matrix.label }}
-        if-no-files-found: error
-  build-binary-windows:
-    runs-on: windows-latest
-    steps:
-    - uses: actions/checkout@v2
-    - uses: ilammy/msvc-dev-cmd@v1.4.1
-    - uses: DeLaGuardo/setup-graalvm@8bbfe44ef9c6f5c07e5af036a1bffd561c037d18
-      with:
-        graalvm-version: '20.3.0.java11'
-    - run: Invoke-Expression -Command "$Env:JAVA_HOME/bin/gu install native-image"
-    - run: .\mvnw.cmd package -Dnative -DskipTests -D cbi.jarsigner.skip=true
-    - run: mv org.eclipse.lemminx\target\lemminx-*.exe lemminx-$(git rev-parse --short "$Env:GITHUB_SHA")-win32.exe
-    - uses: actions/upload-artifact@v2
-      with:
-        name: lemminx-win32
-        path: lemminx-*-win32.exe
-        if-no-files-found: error
+  name: native-image
+  on: [push, pull_request]
+  jobs:
+    build-binary-unix:
+      runs-on: ${{ matrix.os }}
+      strategy:
+        fail-fast: true
+        matrix:
+          os: [macos-latest, ubuntu-latest]
+          include:
+            - os: macos-latest
+              label: 'darwin'
+            - os: ubuntu-latest
+              label: 'linux'
+      steps:
+      - uses: actions/checkout@v2
+      - name: Cache GraalVM and Maven dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            /opt/hostedtoolcache/GraalVM
+            ~/hostedtoolcache/GraalVM
+            ~/.m2/repository
+            ~/.m2/wrapper
+            !~/.m2/repository/org/eclipse/lemminx
+          key: ${{ runner.os }}-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-node
```

Comment:
Just curious about this value. Looking at https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#matching-a-cache-key, this would match any cache key starting with ${{ runner.os }}-node and the most recent if there are multiple , but there aren't any that contain the "node" in the key declaration. Is it some pre-defined key somewhere else ?

Reply:
@fbricon , I think this should just be `${{ runner.os }}- according to the docs. Let me know if that makes sense. I wouldn't mind trying this out on on JDT-LS as well.